validator_client/validator_services/src/sync.rs

```diff
@@ -607,187 +606,81 @@ pub async fn fill_in_aggregation_proofs<S: ValidatorStore, T: SlotClock + 'stati
-                    let proof_slot = slot - 1;
+            // Construct proof for prior slot.
+            let proof_slot = slot - 1;
 
-                    let proof =
-                        make_sync_selection_proof(duties_service_ref, duty, proof_slot, *subnet_id)
+            // Calling the make_sync_selection_proof will return a full selection proof
+            for &subnet_id in &subnet_ids {
+                let duties_service = duties_service.clone();
+                futures_unordered.push(async move {
+                    let result =
+                        make_sync_selection_proof(&duties_service, duty, proof_slot, subnet_id)
                             .await;
 
-                    match proof {
-                        Some(proof) => match proof.is_aggregator::<S::E>() {
-                            Ok(true) => {
-                                debug!(
-                                    validator_index = duty.validator_index,
-                                    slot = %proof_slot,
-                                    %subnet_id,
-                                    "Validator is sync aggregator"
-                                );
-                                Some(((proof_slot, *subnet_id), proof))
-                            }
-                            Ok(false) => None,
-                            Err(e) => {
-                                warn!(
-                                    pubkey = ?duty.pubkey,
-                                    slot = %proof_slot,
-                                    error = ?e,
-                                    "Error determining is_aggregator"
-                                );
-                                None
-                            }
-                        },
-
-                        None => None,
-                    }
+                    result.map(|proof| (duty.validator_index, proof_slot, subnet_id, proof))
                 });
-
-                // Execute all the futures in parallel, collecting any successful results.
-                let proofs = join_all(futures)
-                    .await
-                    .into_iter()
-                    .flatten()
-                    .collect::<Vec<_>>();
-
-                validator_proofs.push((duty.validator_index, proofs));
             }
+        }
 
-            // Add to global storage (we add regularly so the proofs can be used ASAP).
+        while let Some(result) = futures_unordered.next().await {
+            let Some((validator_index, proof_slot, subnet_id, proof)) = result else {
+                continue;
+            };
             let sync_map = duties_service.sync_duties.committees.read();
             let Some(committee_duties) = sync_map.get(&sync_committee_period) else {
-                debug!(period = sync_committee_period, "Missing sync duties");
+                debug!("period" = sync_committee_period, "Missing sync duties");
```

Comment:
I know the original code has the double quote but it's not necessary to have that
```suggestion
                debug!(period = sync_committee_period, "Missing sync duties");
```